### PR TITLE
Minor imprs

### PR DESCRIPTION
**Pull request by kpeluso**
* Remove unnecessary rollback code to stake keeper functions
* Add check that messages aren't nil throughout msg server
* Move "stake is 0" check with error to outside keeper
* added missing topic existence checks

### CODE REVIEW

**Review by br4e — APPROVED** [review]
All of this seems straightforward and looks good to me!

File: x/emissions/keeper/keeper.go [hunk 1]
```diff
@@ -1155,10 +1155,8 @@ func (k *Keeper) RemoveStake(
 	topicId TopicId,
 	reputer sdk.AccAddress,
 	stake Uint) error {
-	// Run checks to ensure that the stake can be removed, and then update the types all at once, applying rollbacks if necessary
-
 	if stake.IsZero() {
-		return errors.New("stake must be greater than zero")
+		return nil
 	}
```
Comment (guilherme-brandao) [inline x/emissions/keeper/keeper.go:1160]: So we gonna keep this `nil`?

File: x/emissions/keeper/keeper.go [hunk 2]
```diff
@@ -1289,34 +1272,21 @@ func (k *Keeper) RemoveDelegateStake(
...
 	stakeUponReputerNew := stakeUponReputer.Sub(stake)
 
-	// types updates -- done all at once after all checks
-
 	// Set new stake from delegator
 	if err := k.SetStakeFromDelegator(ctx, topicId, delegator, stakeFromDelegatorNew); err != nil {
+		fmt.Println("Setting stake from delegator failed")
 		return err
```
Comment (guilherme-brandao) [inline x/emissions/keeper/keeper.go:1278]: We will need to revisit these Prints at some point, not sure how this is useful apart from internal tests.

File: x/emissions/keeper/msgserver/msg_server_registrations.go [hunk 3]
```diff
@@ -12,6 +12,10 @@ import (
 
 // Registers a new network participant to the network for the first time
 func (ms msgServer) Register(ctx context.Context, msg *types.MsgRegister) (*types.MsgRegisterResponse, error) {
+	if msg == nil {
+		return nil, types.ErrReceivedNilRequest
+	}
+
```
Comment (guilherme-brandao) [inline x/emissions/keeper/msgserver/msg_server_registrations.go:18]: prob not needed

File: x/emissions/keeper/msgserver/msg_server_registrations.go [hunk 4]
```diff
@@ -74,6 +78,10 @@ func (ms msgServer) Register(ctx context.Context, msg *types.MsgRegister) (*type
 
 // Remove registration from a topic
 func (ms msgServer) RemoveRegistration(ctx context.Context, msg *types.MsgRemoveRegistration) (*types.MsgRemoveRegistrationResponse, error) {
+	if msg == nil {
+		return nil, types.ErrReceivedNilRequest
+	}
+
```
Comment (guilherme-brandao) [inline x/emissions/keeper/msgserver/msg_server_registrations.go:84]: prob not needed + other places where you added this

File: x/emissions/keeper/keeper.go (hunk 2 above)
Comment (guilherme-brandao) [reply x/emissions/keeper/keeper.go:1278]: prob a better strategy is wrapping the errors instead `errors.Wrap()` from `cosmossdk.io/errors` lib

File: x/emissions/keeper/msgserver/msg_server_registrations.go (hunk 4 above)
Comment (kpeluso) [reply x/emissions/keeper/msgserver/msg_server_registrations.go]: removed

File: x/emissions/keeper/msgserver/msg_server_registrations.go (hunk 3 above)
Comment (kpeluso) [reply x/emissions/keeper/msgserver/msg_server_registrations.go]: removed

File: x/emissions/keeper/msgserver/msg_server_losses.go [hunk 5]
```diff
@@ -19,6 +19,19 @@ func (ms msgServer) InsertBulkReputerPayload(
 	ctx context.Context,
 	msg *types.MsgInsertBulkReputerPayload,
 ) (*types.MsgInsertBulkReputerPayloadResponse, error) {
+	if msg == nil {
+		return nil, types.ErrReceivedNilRequest
+	}
```
[the remark this answers is not in the document]
Comment (kpeluso) [reply x/emissions/keeper/msgserver/msg_server_losses.go]: removed

File: x/emissions/keeper/keeper.go (hunk 1 above)
Comment (kpeluso) [reply x/emissions/keeper/keeper.go:1160]: as we talked ab on slack -- should be an "app layer" decision wrt what to do about throwing or not

File: x/emissions/keeper/keeper.go (hunk 2 above)
Comment (kpeluso) [reply x/emissions/keeper/keeper.go:1278]: if the chain suddenly halts, I think we're gonna be super happy we have them

**Pull request merged**